sdk/include/pti/pti_sync_callback.h

```diff
@@ -0,0 +1,227 @@
+
+typedef enum _pti_gpu_operation_kind {
+  PTI_GPU_OPERATION_KIND_INVALID         = 0,
+  PTI_GPU_OPERATION_KIND_KERNEL          = 1,
+  PTI_GPU_OPERATION_KIND_MEMORY          = 2,
+  PTI_GPU_OPERATION_KIND_OTHER           = 3,
+
+  PTI_OPERATION_KIND_MAX             = 0x7fffffff
+} pti_gpu_operation_kind;
+
+typedef struct _pti_gpu_op_details {
+  pti_gpu_operation_kind             _operation_kind; //<! Kind of the operation: kernel, mem op
+  uint64_t                           _operation_id;   //<! GPU kernel or memory operation instance ID,
+                                                      //<! unique throughout the process
+  uint64_t                           _kernel_handle;  //!< a handle uniquly identifying kernel object as
+                                                      //!< contained in the module at the specific offset
+                                                      //!< it will be zero in case of not defined or
+                                                      //!< for a memory operations
+  const char*                        _name;           //!< symbolic name of a kernel or memcpy operation
+} pti_gpu_op_details;
+
+typedef struct _pti_callback_gpu_op_data {
+  pti_callback_domain             _domain;            // domain of the callback
+  pti_backend_command_list_type   _cmd_list_properties; //!< Immediate command list, Command Queue execute,..
+  pti_backend_command_list_t      _cmd_list_handle;      //!< Device back-end command list handle
+  pti_backend_queue_t             _queue_handle;      //!< Device back-end queue handle
+  pti_device_handle_t             _device_handle;     //!< Device handle,
+  pti_callback_phase              _phase;            //!< Could be ONLY_NOTIFY or API Call ENTER/EXIT
+  uint32_t                        _return_code;       // will be valid only for L0 API EXIT, for others will be nullptr
+  uint32_t                  _correlation_id;    // ID that corresponds to the same call reported by View API records
+  uint32_t                  _operation_count;   // number of operations submitted to the GPU,
+                                                // non-zero only for DOMAIN_DRIVER_GPU_OPERATION_LAUNCH,
+                                                // or API related to GPU operation submission
+  pti_gpu_op_details*       _operation_details; // not Null only for DOMAIN_DRIVER_GPU_OPERATION_LAUNCH,
+                                                // or API related to GPU operation submission.
+} pti_callback_gpu_op_data;
+
+typedef struct _pti_internal_callback_data {
+  pti_callback_domain  _domain;       // domain of the callback
+  pti_callback_phase   _phase;        // THREAD START/END or INTERNAL EVENT
+  uint32_t             _detail;       // depending on the domain should be casted/interpreted
+                                      // as a purpose of an internal PTI thread or
+                                      // pti_internal_event_type
```

Comment:
```suggestion
  uint32_t             _detail;       // For THREAD START/END, this describes the internal PTI thread purpose.
                                      // For INTERNAL EVENT, this describes the pti_internal_event_type.
```

Is there any way to get the internal PTI thread purpose?
Having a number without any actual meaning feels a bit pointless, even if we have the message in the same struct.